File: src/Neuralm.Presentation.Web/src/interfaces/IUserService.ts
import AuthenticateRequest from '@/messages/requests/AuthenticateRequest';
import AuthenticateResponse from '@/messages/responses/AuthenticateResponse';
import RegisterRequest from '@/messages/requests/RegisterRequest';
import RegisterResponse from '@/messages/responses/RegisterResponse';

/**
 * Represents the user service interface.
 */
export default interface IUserService {
  /**
   * Logs in the user with the given user name and password.
   * @param authenticateRequest The authenticate request.
   * @returns The authenticate response.
   */
  login(authenticateRequest: AuthenticateRequest): Promise<AuthenticateResponse>;

  /**
   * Register the user.
   * @param user The user.
   * @returns A value whether the user was succesfully registered.
   */
  register(registerRequest: RegisterRequest): Promise<RegisterResponse>;
}

File: src/Neuralm.Services/Neuralm.Services.UserService/Neuralm.Services.UserService.Application/Interfaces/IUserService.cs
using Neuralm.Services.Common.Application.Interfaces;
using Neuralm.Services.UserService.Messages;
using Neuralm.Services.UserService.Messages.Dtos;
using System.Threading.Tasks;

namespace Neuralm.Services.UserService.Application.Interfaces
{
    /// <summary>
    /// Represents the <see cref="IUserService"/> interface.
    /// </summary>
    public interface IUserService : IService<UserDto>
    {
        /// <summary>
        /// Authenticates a user.
        /// </summary>
        /// <param name="authenticateRequest">The authenticate request.</param>
        /// <returns>Returns an awaitable <see cref="Task"/> with parameter type <see cref="AuthenticateResponse"/>.</returns>
        Task<AuthenticateResponse> AuthenticateAsync(AuthenticateRequest authenticateRequest);

        /// <summary>
        /// Registers a user.
        /// </summary>
        /// <param name="registerRequest">The register request.</param>
        /// <returns>Returns an awaitable <see cref="Task"/> with parameter type <see cref="RegisterResponse"/>.</returns>
        Task<RegisterResponse> RegisterAsync(RegisterRequest registerRequest);
    }
}
